wsagent/che-core-api-git-shared/src/main/java/org/eclipse/che/api/git/shared/EditedRegion.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.git.shared;
+
+import org.eclipse.che.dto.shared.DTO;
+
+/**
+ * Represents edited region of the changed file.
+ *
+ * @author Igor Vinokur
+ */
+@DTO
+public interface EditedRegion {
+
+  /** First line of the edited region. */
+  int getBeginLine();
+
+  void setBeginLine(int startLine);
+
+  EditedRegion withBeginLine(int startLine);
+
+  /** Last line of the edited region. */
+  int getEndLine();
+
+  void setEndLine(int endLine);
+
+  EditedRegion withEndLine(int endLine);
+
+  /** Status of the edition e.g. insertion, modification, deletion. */
+  Type getType();
+
+  void setType(Type type);
+
+  EditedRegion withType(Type type);
+
+  enum Type {
```

Comment:
This is a DTO and it is not good to use it further than Service classes, so I added `che-core-ide-api/src/main/java/org/eclipse/che/ide/api/vcs/EditType.java` to use it in other classes.